### DISCUSSION TITLE
httpx depends on h2 (hyper), which is unmaintained

### DISCUSSION THREAD

**Discussion by apteryks** [encode/httpx#1477] [General]
## Describe the bug

Hello; that's perhaps a misunderstanding on my part, but FWIU this project absorbed hyper (h2) and had it deprecated.  Yet httpx's setup.py still references to h2 (version 3!) as an extra dependency (https://github.com/encode/httpx/blob/0f280af8b170ed5cc48c12a894f71a8b5762f748/setup.py#L65), which in the best case is a one year old release of an now unmaintained library.

What is is useful for? Can it simply be removed? What about hyper compatibility feature with requests (HTTP20Adapter) ? Is there something like this in HTTPX?  Thank you, and sorry for all the questions :-).

**Comment by apteryks** [comment]
Ah, I think I now see the subtlety: hyper (the client) is abandoned, but hyper-h2 (https://github.com/python-hyper/hyper-h2), which is a "pure-Python HTTP/2 protocol stack", is not.

Perhaps it'd be clearer if h2 was brought under this project umbrella.  As it stands, it inherits the hyper branding, which can mislead us to the dead repo (client).

**Comment by florimondmanca** [comment]
@Apteryks I'll conver this to a discussion, if that's OK. :-)

**Reply by florimondmanca** [reply]
@Apteryks Indeed, as far as I understand `h2` is still alive and kicking over at https://github.com/python-hyper/hyper-h2. The project hasn't seen a commit in a while, but that's not a problem since it's mostly in a "done" state.

I believe there was past discussion about moving some of the core Python HTTP libraries to https://github.com/python-http. But AFAIK, the Python HTTP working group project has been discontinued.

**Reply by Kriechi** [reply]
Hey! Maintainer of some of some of the projects in the https://github.com/python-hyper/ organisation here.
Yes, indeed https://github.com/python-hyper/hyper-h2 is still alive - in fact I recently cut the v4 release just a few months ago.

We already started discussing to rename the project to avoid further confusion: https://github.com/python-hyper/hyper-h2/issues/1245
Please drop a 👍  reaction on the top comment over there if you agree.
